_test/test_sqw/test_mask.m

```diff
@@ -0,0 +1,142 @@
+classdef test_mask < TestCase
+
+properties
+    this_dir = fileparts(mfilename('fullpath'));
+    old_warn_state;
+
+    sqw_2d_file_path = '../test_sqw_file/sqw_2d_1.sqw';
+    sqw_2d;
+    sqw_2d_paged;
+    idxs_to_mask;
+    mask_array_2d;
+    masked_2d;
+    masked_2d_paged;
+end
+
+methods
+
+    function obj = test_mask(~)
+        obj = obj@TestCase('test_mask');
+
+        addpath(fullfile(obj.this_dir, 'utils'));
+
+        % Swallow any warnings for when pixel page size set too small
+        obj.old_warn_state = warning('OFF', 'PIXELDATA:validate_mem_alloc');
+
+        obj.sqw_2d = sqw(obj.sqw_2d_file_path);
```

Comment:
does masking work correctly in higher dimensions? This is a 2d test case.

Reply:
The masking routine should be dimension agnostic; if it works in 2D it will work in higher dimensions. 

At the `sqw` level as long as the `mask_array` argument to `mask` has the same number of dimensions as the image data it will work - an error is thrown otherwise. At the `PixelData` level, the `npix` argument is flattened using `repelem` as the array of pixels is flat, so any image dimensionality information is discarded.

Reply:
"The masking routine should be dimension agnostic" the purpose of the tests is to show it _is_ not that is _should be_. It's possible to call repmat incorrectly, especially in higher dimensions.

Even if there are `PixelData` level tests that prove this works, an additional test at `SQW` level would confirm all the wiring is correct between the layers and give a code example of higher-dimension calls.